Web/LearningStarter/learning-starter-web/src/components/navigation/navigation.tsx

```diff
@@ -23,129 +40,188 @@ type NavigationItem = {
     }
-  | { nav?: never; children: NavigationItem[] }
+  | { nav?: never; children: NavigationItemForceNav[] }
 );
 
-//This is where the navigation buttons are defined.
+export type NavigationItemForceNav = {
+  text: string;
+  icon?: IconProp | undefined;
+  hide?: boolean;
+  nav: NavLinkProps;
+};
+
+const navigation: NavigationItem[] = [
+  {
+    text: "Home",
+    hide: false,
+    nav: {
+      to: routes.home,
+    },
+  },
+  {
+    text: "User",
+    hide: false,
+    nav: {
+      to: routes.user,
+    },
+  },
+];
+
 const DesktopNavigation = () => {
-  const navigation: NavigationItem[] = useMemo(() => {
-    return [
-      {
-        text: "Home",
-        icon: "home",
-        hide: false,
-        nav: {
-          to: routes.home,
-        },
-      },
-      {
-        text: "User",
-        icon: "user",
-        hide: false,
-        nav: {
-          to: routes.user,
-        },
-      },
-    ];
-  }, []);
+  const { classes, cx } = useStyles();
+  const { pathname } = useLocation();
+  const [active, setActive] = useState(navigation[0].nav?.to.toString());
+
+  useEffect(() => {
+    setActive(pathname);
+  }, [pathname, setActive]);
 
-  //This is where the navigation buttons are mapped over to produce the links and such.
   return (
-    <Menu
-      secondary
-      role="navigation"
-      className="desktop-navigation"
-      size="large"
-    >
-      {navigation
-        .filter((x) => !x.hide)
-        .map((x, i) => {
-          if (x.children) {
-            return (
-              <Dropdown
-                key={i}
-                trigger={
-                  <span>
-                    {x.icon && <Icon size="small" fitted name={x.icon} />}{" "}
-                    {x.text}
-                  </span>
-                }
-                pointing
-                className="link item"
-              >
-                <Dropdown.Menu>
-                  {x.children
-                    .filter((x) => !x.hide)
-                    .map((y) => {
-                      return (
-                        <Dropdown.Item
-                          key={`${y.text}`}
-                          as={NavLink}
-                          to={y.nav?.to}
-                        >
-                          {y.icon && <Icon size="small" fitted name={y.icon} />}{" "}
-                          {y.text}
-                        </Dropdown.Item>
-                      );
-                    })}
-                </Dropdown.Menu>
-              </Dropdown>
-            );
-          }
-          return (
-            <Menu.Item key={i} as={NavLink} {...x.nav}>
-              {x.icon && <Icon size="small" name={x.icon} />} {x.text}
-            </Menu.Item>
-          );
-        })}
-    </Menu>
+    <>
+      <Container px={0} className={classes.desktopNav}>
+        <Flex direction="row" align="center" className={classes.fullHeight}>
+          {navigation
+            .filter((x) => !x.hide)
+            .map((x, i) => {
```

Comment:
***NIT*** I feel like you could name these a little more explicitly ex: `item, index`